typescript/src/deposit-sweep.ts

```diff
@@ -130,238 +150,297 @@ export async function assembleDepositSweepTransaction(
   }
-  // Build script and set it as input's witness
-  transaction.scriptInput(inputIndex, previousOutput, walletKeyRing)
-  // Build signature and add it in front of script in input's witness
-  transaction.signInput(inputIndex, previousOutput, walletKeyRing)
+
+  const sigHashType = Transaction.SIGHASH_ALL
+
+  if (isP2PKH(prevOutScript)) {
+    // P2PKH
+    const sigHash = transaction.hashForSignature(
+      inputIndex,
+      prevOutScript,
+      sigHashType
+    )
+
+    const signature = script.signature.encode(
+      keyPair.sign(sigHash),
+      sigHashType
+    )
+
+    const scriptSig = payments.p2pkh({
+      signature: signature,
+      pubkey: keyPair.publicKey,
+    }).input!
+
+    transaction.ins[inputIndex].script = scriptSig
+  } else if (isP2WPKH(prevOutScript)) {
+    // P2WPKH
+    const decompiledScript = script.decompile(prevOutScript)
+    if (
+      !decompiledScript ||
+      decompiledScript.length !== 2 ||
+      decompiledScript[0] !== 0x00 ||
+      !Buffer.isBuffer(decompiledScript[1]) ||
+      decompiledScript[1].length !== 20
+    ) {
+      throw new Error("Invalid script format")
+    }
+
+    const publicKeyHash = decompiledScript[1]
+    const p2pkhScript = payments.p2pkh({ hash: publicKeyHash }).output!
+
+    const sigHash = transaction.hashForWitnessV0(
+      inputIndex,
+      p2pkhScript,
+      prevOutValue,
+      sigHashType
+    )
+
+    const signature = script.signature.encode(
+      keyPair.sign(sigHash),
+      sigHashType
+    )
+
+    transaction.ins[inputIndex].witness = [signature, keyPair.publicKey]
+  } else {
+    throw new Error("Unknown type of main UTXO")
+  }
 }
 
 /**
- * Creates and sets `scriptSig` for the transaction input at the given index by
- * combining signature, wallet public key and deposit script.
- * @param transaction - Mutable transaction containing the input to be signed.
- * @param inputIndex - Index that points to the input to be signed.
- * @param deposit - Data of the deposit.
- * @param walletKeyRing - Key ring created using the wallet's private key.
- * @returns Empty promise.
+ * Signs a P2SH deposit transaction input and sets the `scriptSig`.
+ * @param transaction - The transaction containing the input to be signed.
+ * @param inputIndex - Index pointing to the input within the transaction.
+ * @param deposit - Details of the deposit transaction.
+ * @param prevOutValue - The value from the previous transaction output.
+ * @param keyPair - A Signer object with the public and private key pair.
+ * @returns An empty promise upon successful signing.
  */
 async function signP2SHDepositInput(
-  transaction: any,
+  transaction: Transaction,
   inputIndex: number,
   deposit: Deposit,
-  walletKeyRing: any
-): Promise<void> {
-  const { walletPublicKey, depositScript, previousOutputValue } =
-    await prepareInputSignData(transaction, inputIndex, deposit, walletKeyRing)
+  prevOutValue: number,
+  keyPair: Signer
```

Comment:
`walletKeyPair`?

Reply:
Done in 59fb549db1ebf7f92310cd5c1ac05dbdf9f34630.